docs/topics/gradle/gradle-configure-project.md

```diff
@@ -386,6 +386,46 @@ integrationTestCompilation {
 Here, the `integrationTest` compilation is associated with the `main` compilation that gives access to `internal`
 objects from functional tests.
 
+### Configure with Java JPMS enabled
+
+To make the Kotlin Gradle plugin work with [JPMS](https://www.oracle.com/corporate/features/understanding-java-9-modules.html), 
+add the following lines to your build script and replace `YOUR_MODULE_NAME` with the name of your module:
+
+<tabs group="build-script">
+<tab title="Kotlin" group-key="kotlin">
+        
+```kotlin
+tasks.named("compileJava", JavaCompile::class.java) {
+    options.compilerArgumentProviders.add(CommandLineArgumentProvider {
+        listOf( 
+            // Workaround to add classpath into modules for Java compilation so automatic modules can work
+            "--module-path", classpath.asPath,
+            // Provide compiled Kotlin classes to javac – needed for Java/Kotlin mixed sources to work
+            "--patch-module", "YOUR_MODULE_NAME=${sourceSets["main"].output.asPath}"
```

Comment:
Can such way of patching mask some problems when all outputs including Java are passed again to the input of Java compilation?

Reply:
Usually such compilation outputs contain compiled `.class` files and are passed via `--classpath` option for another Java compilation. Why do you think it could mask some problems? :thinking:

Reply:
Are these only Kotlin class files directory or Kotlin + Java class files directories?

Reply:
Both - consider mixed Kotlin/Java project output as a single artifact containing all Kotlin and Java class files and resources.